src/node_messaging.h

```diff
@@ -15,6 +15,29 @@ class MessagePort;
 
 typedef MaybeStackBuffer<v8::Local<v8::Value>, 8> TransferList;
 
+// Used to represent the in-flight structure of an object that is being
+// transfered or cloned using postMessage().
+class TransferData : public MemoryRetainer {
+ public:
+  // Deserialize this object on the receiving end after a .postMessage() call.
+  // - `context` may not be the same as `env->context()`. This method should
+  //    not produce JS objects coming from Contexts other than `context`.
+  // - `self` is a unique_ptr for the object that this is being called on.
+  // - The return value is treated like a `Maybe`, i.e. if `nullptr` is
+  //   returned, any further deserialization of the message is stopped and
+  //   control is returned to the event loop or JS as soon as possible.
+  virtual BaseObjectPtr<BaseObject> Deserialize(
+      Environment* env,
+      v8::Local<v8::Context> context,
+      std::unique_ptr<TransferData> self) = 0;
```

Comment:
@tniessen You will probably also want to include f172c2dc874a9f9575bd958e0d3eb828ced224ce in your PR. I missed the fact that returning a plain `BaseObject*` might not work in weird edge case situations, because objects like `KeyObject` or `FileHandle` are weak objects, and only holding a `BaseObject*` does not prevent them from being garbage-collected between returning them from `Deserialize()` and adding the corresponding JS object to the message data object.